pybamm/solvers/idaklu_solver.py

```diff
@@ -711,3 +711,87 @@ def jaxify(
             calculate_sensitivities=calculate_sensitivities,
         )
         return obj
+
+    def get_termination_reason(
+        self, solution: pybamm.Solution, events: dict, variables: dict
+    ):
+        """
+        See `pybamm.BaseSolver.get_termination_reason()`
+        Overwrites the default method to handle events when output_variables are
+        provided to the IDAKLU solver, so only a subset of the model state vector is
+        returned.
+
+        Parameters
+        ----------
+        solution : :class:`pybamm.Solution`
+            The solution object
+        events : dict
+            Dictionary of events
+        variables : dict
+            Dictionary of model variables
+        """
+        if not self.output_variables or solution.termination == "final time":
+            return super().get_termination_reason(solution, events)
+
+        if solution.termination == "event":
+            pybamm.logger.debug("Start post-processing events")
+
+            termination_events = [
+                x for x in events if x.event_type == pybamm.EventType.TERMINATION
+            ]
+
+            # Get final event value
+            final_event_values = {}
+
+            output_vars = {var: variables[var] for var in self.output_variables}
+
+            for event in termination_events:
+                event_children = event.expression.children
+
+                # find the output variable that matches one of the event children
+                matching_var = {
+                    k: v for k, v in output_vars.items() if v in event_children
+                }
+                if len(matching_var) != 1:
+                    try:
+                        final_event_values[event.name] = event.expression.evaluate(
+                            solution.t_event,
+                            solution.y_event,
+                            inputs=solution.all_inputs[-1],
+                        )
+                        continue
+                    except ValueError as e:  # pragma: no cover
```

Comment:
Likewise, is this cause for a critical failure? I would consider replacing with a warning.

Reply:
See above